app/charts/index.ts

```diff
@@ -1153,17 +1202,18 @@ const convertTableFieldsToSegmentField = ({
 
   if (component) {
     const { componentIri } = component;
+    const actualComponent = [...dimensions, ...measures].find(
+      (d) => d.iri === componentIri
+    ) as DimensionMetadataFragment;
+    const hasDefaultColors = hasDimensionDefaultColors(actualComponent);
+    const palette = hasDefaultColors ? "default" : DEFAULT_PALETTE;
```

Comment:
Nit: Maybe a getDefaultPalette(component) function would be helpful there as the check + assignment is repeated for every chart.